testing/python/fixture.py

```diff
@@ -2547,6 +2548,44 @@ def test_foo(fix):
             '*test_foo*alpha*',
             '*test_foo*beta*'])
 
+    @pytest.mark.issue920
+    def test_deterministic_fixture_collection(self, testdir):
+        a = testdir.mkdir("a")
```

Comment:
Not sure you need the sub directory, plus you can ask `testdir` to create the file for you:

```python
testdir.makepyfile(''''
    import pytest

    @pytest.fixture(scope="module",
    ...
```

This will create a test file named `test_deterministic_fixture_collection.py` (same as the test name) in the test directory.